File: src/main/java/com/kelsos/mbrc/commands/ConnectionStatusChangedCommand.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.data.SocketMessage;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.messaging.NotificationService;
import com.kelsos.mbrc.model.MainDataModel;
import com.kelsos.mbrc.others.Protocol;
import com.kelsos.mbrc.services.ProtocolHandler;
import com.kelsos.mbrc.services.SocketService;
import com.squareup.otto.Bus;

public class ConnectionStatusChangedCommand implements ICommand
{
	@Inject
	MainDataModel model;
    @Inject
    SocketService service;
	@Inject
	NotificationService notificationService;

	public void execute(IEvent e)
	{
		model.setConnectionState(e.getDataString());
		if(model.getIsConnectionActive()){
            service.sendData(new SocketMessage(Protocol.Player, Protocol.Request, "Android"));
		} else {
			notificationService.cancelNotification(NotificationService.NOW_PLAYING_PLACEHOLDER);
		}
	}
}

File: app/src/main/java/com/kelsos/mbrc/commands/ConnectionStatusChangedCommand.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.common.state.ConnectionModel
import com.kelsos.mbrc.networking.client.SocketMessage
import com.kelsos.mbrc.networking.client.SocketService
import com.kelsos.mbrc.networking.protocol.Protocol
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage
import com.kelsos.mbrc.platform.mediasession.SessionNotificationManager
import javax.inject.Inject

class ConnectionStatusChangedCommand
  @Inject
  constructor(
    private val model: ConnectionModel,
    private val service: SocketService,
    private val sessionNotificationManager: SessionNotificationManager,
  ) : ProtocolAction {
    override fun execute(message: ProtocolMessage) {
      model.setConnectionState(message.dataString)

      if (model.isConnectionActive) {
        service.sendData(SocketMessage.create(Protocol.PLAYER, "Android"))
      } else {
        sessionNotificationManager.cancelNotification(SessionNotificationManager.NOW_PLAYING_PLACEHOLDER)
      }
    }
  }
